File: GO/20.go
func isValid(s string) bool {
	if len(s) <= 1 {
		return false
	}
	stack := []byte{}
	stack = append(stack, s[0])
	for i := 1; i < len(s); i++ {
		if len(stack) == 0 || s[i] == '(' || s[i] == '[' || s[i] == '{' {
			stack = append(stack, s[i])
		} else {
			if s[i] == ')' && stack[len(stack)-1] != '(' {
				return false
			}
			if s[i] == ']' && stack[len(stack)-1] != '[' {
				return false
			}
			if s[i] == '}' && stack[len(stack)-1] != '{' {
				return false
			}
			stack = stack[:len(stack)-1]
		}
	}
	return len(stack) == 0
}

File: 20.java
class Solution {
    public boolean isValid(String s) {
        Deque<Character> stack = new ArrayDeque<>();
        for(int i = 0;i < s.length();i ++) {
            char c = s.charAt(i);
            if(c == '(' || c == '[' || c == '{') {
                stack.push(c);
            }else if(c == ')') {
                if(stack.size() != 0 && stack.peek() == '(') stack.pop();
                else return false;
            }else if(c == ']') {
                if(stack.size() != 0 && stack.peek() == '[') stack.pop();
                else return false;
            }else {
                if(stack.size() != 0 && stack.peek() == '{') stack.pop();
                else return false;
            }
        }
        return stack.size() == 0;
    }
}